net/tstun/wrap_test.go

```diff
@@ -608,191 +608,222 @@ func TestNATCfg(t *testing.T) {
+					Peers: []wgcfg.Peer{
+						node(peer1IP, selfEIP1),
+						node(peer2IP, selfEIP2, subnet),
+					},
+				},
+				snatMap: map[netip.Addr]netip.Addr{
+					peer1IP:  selfEIP1,
+					peer2IP:  selfEIP2,
+					subnetIP: selfEIP2,
+				},
+				dnatMap: map[netip.Addr]netip.Addr{
+					selfNativeIP: selfNativeIP,
+					selfEIP1:     selfNativeIP,
+					selfEIP2:     selfNativeIP,
+					subnetIP:     subnetIP,
+				},
 			},
-		},
-		{
-			name: "exit-node-require-nat-peer-doesnt",
-			wcfg: &wgcfg.Config{
-				Addresses: selfAddrs,
-				Peers: []wgcfg.Peer{
-					node(peer1IP, noIP),
-					node(peer2IP, selfEIP2, exitRoute),
+			{
+				name: "multiple-peers-require-nat-with-default-route",
+				wcfg: &wgcfg.Config{
+					Addresses: selfAddrs,
+					Peers: []wgcfg.Peer{
+						node(peer1IP, selfEIP1),
+						node(peer2IP, selfEIP2, exitRoute),
+					},
+				},
+				snatMap: map[netip.Addr]netip.Addr{
+					peer1IP:  selfEIP1,
+					peer2IP:  selfEIP2,
+					publicIP: selfEIP2,
+				},
+				dnatMap: map[netip.Addr]netip.Addr{
+					selfNativeIP: selfNativeIP,
+					selfEIP1:     selfNativeIP,
+					selfEIP2:     selfNativeIP,
+					subnetIP:     subnetIP,
 				},
 			},
-			snatMap: map[netip.Addr]netip.Addr{
-				peer1IP:  selfNativeIP,
-				peer2IP:  selfEIP2,
-				publicIP: selfEIP2,
+			{
+				name: "no-nat",
+				wcfg: &wgcfg.Config{
+					Addresses: selfAddrs,
+					Peers: []wgcfg.Peer{
+						node(peer1IP, noIP),
+						node(peer2IP, noIP),
+					},
+				},
+				snatMap: map[netip.Addr]netip.Addr{
+					peer1IP:  selfNativeIP,
+					peer2IP:  selfNativeIP,
+					subnetIP: selfNativeIP,
+				},
+				dnatMap: map[netip.Addr]netip.Addr{
+					selfNativeIP: selfNativeIP,
+					selfEIP1:     selfEIP1,
+					selfEIP2:     selfEIP2,
+					subnetIP:     subnetIP,
+				},
 			},
-			dnatMap: map[netip.Addr]netip.Addr{
-				selfNativeIP: selfNativeIP,
-				selfEIP2:     selfNativeIP,
-				subnetIP:     subnetIP,
+			{
+				name: "exit-node-require-nat-peer-doesnt",
+				wcfg: &wgcfg.Config{
+					Addresses: selfAddrs,
+					Peers: []wgcfg.Peer{
+						node(peer1IP, noIP),
+						node(peer2IP, selfEIP2, exitRoute),
+					},
+				},
+				snatMap: map[netip.Addr]netip.Addr{
+					peer1IP:  selfNativeIP,
+					peer2IP:  selfEIP2,
+					publicIP: selfEIP2,
+				},
+				dnatMap: map[netip.Addr]netip.Addr{
+					selfNativeIP: selfNativeIP,
+					selfEIP2:     selfNativeIP,
+					subnetIP:     subnetIP,
+				},
 			},
-		},
-	}
+		}
 
-	for _, tc := range tests {
-		t.Run(tc.name, func(t *testing.T) {
-			ncfg := natV4ConfigFromWGConfig(tc.wcfg)
-			for peer, want := range tc.snatMap {
-				if got := ncfg.selectSrcIP(selfNativeIP, peer); got != want {
-					t.Errorf("selectSrcIP[%v]: got %v; want %v", peer, got, want)
-				}
-			}
-			for dstIP, want := range tc.dnatMap {
-				if got := ncfg.mapDstIP(dstIP); got != want {
-					t.Errorf("mapDstIP[%v]: got %v; want %v", dstIP, got, want)
-				}
+		t.Run(testName, func(t *testing.T) {
```

Comment:
i typically wrap these type of tests like

```
for _, tt := range tests {
  for _, isV6 := range []bool{false, true} {
    testName := fmt.Sprintf("v6=%t/%v", isV6, tt.name)
  }
}
```

Reply:
Unfortunately the `v6` parameter needs to be outside the test cases and not the other way around (bc different IPs are used), but done otherwise.